### ISSUE TITLE
S3 - RequestTimeout during large files

### ISSUE THREAD

**Issue by lloydcotten** [aws/aws-cli#401]
I'm trying to upload a large file (9 GB) and getting a RequestTimeout error using `aws s3 mv ...`

I haven't fully tested it yet, but it seems like if I run the command over and over it will eventually work.

Here's the debug log from a failed attempt: https://s3.amazonaws.com/nimbus-public/s3_backup.log

I'll post back if I determine that retrying the command several times works or not.

aws version: aws-cli/1.1.2 Python/2.7.3 Windows/2008ServerR2

**Comment by naren3k** [comment]
Hi, I m getting this even for 2KB files through data power. However 1KB files works fine.

**Comment by techdragon** [comment]
Still appears to be a problem. Brand new Ubuntu VM, installed AWS CLI tools, 
`aws-cli/1.10.1 Python/3.5.2 Linux/4.4.0-38-generic botocore/1.3.23`

Once more I face the dreaded ConnectionResetError 104!

```
upload failed: 1.pdf to s3://mybucket/1.pdf ('Connection aborted.', ConnectionResetError(104, 'Connection reset by peer'))
upload failed: 1.pdf to s3://mybucket/1.pdf ('Connection aborted.', ConnectionResetError(104, 'Connection reset by peer'))
upload failed: 2.pdf to s3://mybucket/2.pdf ('Connection aborted.', ConnectionResetError(104, 'Connection reset by peer'))
upload failed: 3.jpg to s3://mybucket/3.jpg ('Connection aborted.', ConnectionResetError(104, 'Connection reset by peer'))
```

**Comment by teamkde** [comment]
This happens if using the AWS UI to upload large files too.

**Comment by kyle-sorensen** [comment]
Hey guys, I am seeing this issue too. I get it while trying to do a multipart upload. I thought my chunk size was too large, but alas I changed it to 5MB and it still gets a timeout error.

**Comment by sshaw** [comment]
Same problem. I ended up using good ol' [s3cmd](https://github.com/s3tools/s3cmd):

```
s3cmd --continue-put put 5GB_file s3://some-bucket
```
 Worked like a charm. Who says `s3cmd` is dead? :neckbeard:  📁 ➡️ 🌎 ➡️ 📬

**Comment by tobias-khs** [comment]
Seeing this issue with:

    > aws s3 --version
    > aws-cli/1.11.44 Python/2.7.3 Linux/3.7.10-1.40-desktop botocore/1.5.7

20 GB file. Retrying usually works, but still

**Comment by b-xk** [comment]
Also seeing this with 
> aws s3 --version
> aws-cli/1.11.56 Python/2.7.12+ Linux/4.6.0-kali1-amd64 botocore/1.5.9

using both the S3 Accelerate endpoint, and the non-accelerated endpoint gave the same 'connection aborted' error as described above, even after trying 10-12 times, with a 3Gb file. (us-west-2)

Interestingly/anecdotally:  Removing the following option from the 's3 cp' command allowed the copy to complete on the very next attempt.

Removed the following option;  --sse AES256

This suggests to me that there is a (regression?) issue with the way that the confirmation of the part-completions from the server-side are being handled, which are likely made worse/slower when SSE is enabled.  Looks like the client may be timing out too quickly ?

**Comment by ossie-git** [comment]
This issue isn't specific to large files.  I was just testing something and tried to upload a copy of the WordPress source files.  After about 200 files, it starts to give me the ConnectionResetError(104, 'Connection reset by peer')) for files that are 300KB in size and even smaller

aws s3 --version
aws-cli/1.11.13 Python/3.5.2 Linux/4.4.0-1031-aws botocore/1.4.70

**Comment by ossie-git** [comment]
Just a small update.  It seems to have been fixed in some later version (and even runs a lot faster).  I checked out the latest version using pip install --upgrade awscli so my version was as follows:

aws s3 --version
aws-cli/1.11.138 Python/3.5.2 Linux/4.4.0-1031-aws botocore/1.6.5

Re-ran the command it successfully copied all the files (+ was a lot faster than the default version that you install with Ubuntu 16.04 LTS)

**Comment by dcg9381** [comment]
Thank you for the update..

**Comment by tgirgin23** [comment]
This hasn't been fixed yet (running latest aws-cli). I did get it to work by setting **--cli-read-timeout** to 0.

`aws --cli-read-timeout 0 s3 cp s3://file .`

File successfully cp/mv after doing so.

**Comment by opyh** [comment]
Experiencing the same issue with a 49 GB file - `--cli-read-timeout 0` didn't help here.

```bash
aws s3 --version
aws-cli/1.14.44 Python/3.6.9 Linux/4.19.0-0.bpo.6-amd64 botocore/1.8.48
```

I wonder why the protocol cannot fall back to polling instead of relying on a socket connection being open for a long time while the chunks are being assembled on the server?

**Comment by ham1** [comment]
I had this issue today as well, `--cli-read-timeout 0` did nothing for me.

I "solved" it by using [rclone](https://rclone.org/s3/) instead - it worked flawlessly.

**Comment by mdsp0292** [comment]
I'm facing the same issue as well when trying to upload a zip file of size 800 MiB

An error occurred (RequestTimeout) when calling the UploadPart operation (reached max retries: 2): Your socket connection to the server was not read from or written to within the timeout period. Idle connections will be closed.

aws s3 --version
aws-cli/2.1.37 Python/3.8.8 Linux/4.15.0-123-generic exe/x86_64.ubuntu.18 prompt/off

command used
aws --cli-read-timeout 0 s3 cp

**Comment by asdf01** [comment]
It's coming up to 8 years since this issue was raised.  This issue still exists. Reopen this ticket and fix it!  Jeff Bezos has gone to space already, and we still can't reliably copy large files to AWS s3.  We are sending terabytes of data half way around the world in our repeated attempts to upload a simple 200GB file.  

Just because people are silent on this issue doesn't mean this issue doesn't exist.  It either means they've found their own workaround through 3rd party tools or moved on to other cloud providers.  

AWS purports to be big on security but can't do basic sheet like this.  This forces AWS customers into providing their AWS security credentials to questionable 3rd party tools to achieve the most basic of things like uploading a file.  

AWS is really becoming a joke of a cloud provider when they leave fundamental issues like this unresolved for 7+ years.  I'm becoming a clown of an employee to my employer for continuing to choose AWS amid all these issues.  

Reopen this ticket and fix it.  You guys could have chosen to address this issue in one of many ways.  You could have provided:
- a configurable server timeout parameter 
- a configurable client side retry count
- some type of resume flag that takes the part number 

Don't choose to do nothing like a number of other AWS issues.  We are getting used to celebrating the anniversary of other unresolved AWS issues in the issue comments just to entertain the other issue followers as a joke.  We are certainly not getting any attention from the AWS employees.  

Customers are not going to continually beg for something if they are not getting any attention from AWS.  They will still be talking about AWS, just not on your platforms.  Don't let this situation devolve into a surprise Pikachu face moment for AWS when all of your customers move on to a cloud provider that had obviously necessary features and responds to their customers.  

In summary:
It's ducking 2021, Jeff Bezos is in space already.  We are not asking for flying cars.  Just fix the ducking aws s3 cp retry issue.

**Comment by Wambosa** [comment]
@asdf01, I got some much needed laughing relief from your comment. Its true that AWS has some glaring holes in it while they forge ahead with other solutions.

Its been awhile since I was stung by this, my team is dealing with it now, and unfortunately, we might have to homebrew our own wrapper around it.

I can see that this repo has over 400 issues at time of writing. It must be hard for @jamesls to manage alone. Not sure what is going on back there, as I know we'd have trouble supporting a repo with so many issues. It can be so overwhelming; however I must second the request above to not forget about the architects whom are loyal to AWS, driving millions and millions of dollars to the service each time we build a scaling solution.

It is hard to track how much issues like this cause market share loss, but I promise you it does.

**Comment by tuananh** [comment]
11 years and this issue is still open.

**Comment by juangpc** [comment]
13 years and this issue is still open.
